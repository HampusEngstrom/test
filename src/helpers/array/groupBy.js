export const groupBy = (arr, key) => {
  const result = {};
  for (let i = 0; i < arr.length; i++) {
    const item = arr[i];
    if (item.hasOwnProperty(key)) {
      const value = item[key];
      if (!result[value]) {
        result[value] = [];
      }
      result[value].push(item);
    }
  }
  return result;
};
