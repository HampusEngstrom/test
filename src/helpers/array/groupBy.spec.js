import { groupBy } from './groupBy';
{
}
describe('test groupBy', () => {
  it('returns grouped object', () => {
    const arr = [
      { id: '1', a: 'a1', b: 'b1' },
      { id: '2', a: 'a2', b: 'b2' },
      { id: '3', a: 'a2', b: 'b1' },
      { id: '4', c: 'c1', d: 'd1' },
    ];
    const key = 'b';
    const result = groupBy(arr, key);
    expect(result).toMatchSnapshot();
  });
});
