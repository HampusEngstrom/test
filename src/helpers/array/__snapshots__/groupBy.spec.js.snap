// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test groupBy returns grouped object 1`] = `
Object {
  "b1": Array [
    Object {
      "a": "a1",
      "b": "b1",
      "id": "1",
    },
    Object {
      "a": "a2",
      "b": "b1",
      "id": "3",
    },
  ],
  "b2": Array [
    Object {
      "a": "a2",
      "b": "b2",
      "id": "2",
    },
  ],
}
`;
